### ISSUE TITLE
SSM not using generated or provided self-signed certs (with USE_SSL = true)

### ISSUE THREAD

**Issue by brettcooper** [localstack/localstack#1119]
We are using Kinesis and have USE_SSL set to true in order to communicate properly, but setting USE_SSL = true makes all Local Stack services run under TLS / SSL.  SSM is not using the generated SSL certs from Local Stack, so I even tried to generate my own SSL certs as shown on the README page for Local Stack to force Local Stack to use them in a docker-compose.yml file.  But, SSM still does not use my generated cert, but a dummy cert that I can't seem to change.  Here is the cert being used by SSM:

openssl s_client -showcerts -connect "localhost:4583"
```
CONNECTED(00000005)
depth=0 CN = *, O = Dummy Certificate
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = *, O = Dummy Certificate
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/CN=*/O=Dummy Certificate
   i:/CN=Untrusted Authority/O=Self-Signed
-----BEGIN CERTIFICATE-----
MIIC1zCCAb8CCF9rbsPHJjAAMA0GCSqGSIb3DQEBCwUAMDQxHDAaBgNVBAMME1Vu
dHJ1c3RlZCBBdXRob3JpdHkxFDASBgNVBAoMC1NlbGYtU2lnbmVkMB4XDTE5MDIx
MTE1NTQ1M1oXDTIwMDIxMTE1NTQ1M1owKDEKMAgGA1UEAwwBKjEaMBgGA1UECgwR
RHVtbXkgQ2VydGlmaWNhdGUwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
AQDO4/CENe31YWTA47yseDJOWoW/hVBTlpN9oVmI2w1gIOhLoz5AYL31wKFr8Zmp
M9Xinc/1Cu0V2vVAKtdyFSyQn10+bPkqzWRnrvP8Lulx/gaAG5Viea9MNzDQB0qt
ss6MXXgBAkXeG7U9lRJJfM0TBtk5QaHbAK0GAi9Qpz26/Mw9zEXwt1+3Z4FOEu1b
8CwKVKx+YnxwES6ublIa/jCxhin2spVmI5fqozehLD4d1aBPxoNUiNZwt5XZ2LMK
a4ouxae5cfsYSjtXeioQTih3lKzLcKW6RZzM6HM3ZoVf9KZRlbsXiTEy75sWq5vA
pdYY8aEp6MPYCH6BsskO3goPAgMBAAEwDQYJKoZIhvcNAQELBQADggEBABT/tXkL
ZcXtQqxkAUsnhRHvEYQfp3YbKDgM5JdOqDL6dTnfi2BHYjv3enVi0cTda/apZJ3T
DybjacPlY70h+nDDHuawZnhcQ4B0GtL12N3HNRBhC7PdkzovFmkD6TlzCv/yJn1q
DSRf762O6cau/CQUux0JSmxRl3m0sdPTLWmnzpw5De4193RLmP6wcKf58cm0zaZO
Wo3jFzlfRMcXluFu95V/g2ojcY/3ZjslGYB+YGVS4wtN5YHKWvfV1Q/iAcl0cvf+
Rxm+OfRoGHcXSBHnSEDtZH6g08S9q58Fjn2B6hls1ZT30aTwHJtQ58Vw4Kf5u9/7
9KNTljIPjx1bVgg=
-----END CERTIFICATE-----
---
Server certificate
subject=/CN=*/O=Dummy Certificate
issuer=/CN=Untrusted Authority/O=Self-Signed
---
No client certificate CA names sent
Server Temp Key: ECDH, X25519, 253 bits
---
SSL handshake has read 1342 bytes and written 285 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-CHACHA20-POLY1305
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-CHACHA20-POLY1305
    Session-ID: AED9B35D3C0E1885E0A66967A73E35E2D9563DF6909B400A6A80FCA41C0FD04C
    Session-ID-ctx: 
    Master-Key: 7D710248D2BF4D2A042EC07CED103CB921CEA5F5BDCFE3F7F929FBAC725E1FD3580917409A97AA1A12501D76A29075E1
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - 20 2c 75 f1 48 ff f1 17-a7 78 70 11 fe 72 e5 38    ,u.H....xp..r.8
    0010 - a2 43 d1 4c f0 eb 35 14-d6 85 b2 ce 2c 70 9b 20   .C.L..5.....,p. 
    0020 - af d9 fb b6 2b 32 44 0c-d1 4b 3e fb 38 c1 88 a5   ....+2D..K>.8...
    0030 - 0b b0 97 b1 4f 9a 2d 20-09 89 7d ed e2 a8 5d a1   ....O.- ..}...].
    0040 - 5a ae 59 e9 f3 7e 5d dd-25 ad 60 04 12 e1 33 5c   Z.Y..~].%.`...3\
    0050 - 60 61 2b cb 50 44 5a ed-f1 c6 b6 fb 6f 66 63 1f   `a+.PDZ.....ofc.
    0060 - 89 c8 55 3f fa 65 14 d4-22 7a 5d 45 4b 10 65 07   ..U?.e.."z]EK.e.
    0070 - a7 01 c1 88 07 75 f9 c9-1a 40 38 e5 5d 0e 6b 44   .....u...@8.].kD
    0080 - 38 2a 82 94 12 42 92 2d-f7 c2 79 e5 e6 ca 95 33   8*...B.-..y....3
    0090 - 9e 3f a7 5f 8b 80 aa f4-06 df 63 6c 77 3f 71 cf   .?._......clw?q.

    Start Time: 1549900543
    Timeout   : 7200 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
```

docker-compose.yml:
```
version: '2.1'

services:
  localstack:
    image: localstack/localstack
    ports:
      - "4567-4584:4567-4584"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES:-dynamodb, kinesis, sqs, ssm, cloudwatch}
      - DEBUG=${DEBUG-1}
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - DEFAULT_REGION=${DEFAULT_REGION:-us-west-2}
      - USE_SSL=${USE_SSL:-true}
      - FORCE_NONINTERACTIVE=${FORCE_NONINTERACTIVE:-true}
    volumes:
      - "${PWD}/certs:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

I believe this is a bug and SSM should be using the generated SSL certs from Local Stack, or the certs I created via a docker-compose.yml.

**Comment by whummer** [comment]
Thanks for reporting @brettcooper . This should be fixed in #1176 - please re-open this issue if the problem persists. Thanks

**Comment by brettcooper** [comment]
Verified fix is working.  Thank you!
